content/docs/userspace/full-stack/3-agent.md

```diff
@@ -0,0 +1,335 @@
+      =/  end=@  (add 1 (rash i.t.t.t.t.path dem))
+      :^  ~  ~  %journal-update
+      !>  ^-  update
+      [now %jrnl (tap:j-orm (lot:j-orm journal `end `start))]
+    ==
+  ::
+      [%x %updates *]
+    ?+    t.t.path  (on-peek:def path)
+        [%all ~]
+      :^  ~  ~  %journal-update
+      !>  ^-  update
+      [now %logs (tap:log-orm log)]
+    ::
+        [%since @ ~]
+      =/  since=@  (rash i.t.t.t.path dem)
+      :^  ~  ~  %journal-update
+      !>  ^-  update
+      [now %logs (tap:log-orm (lot:log-orm log `since ~))]
+    ==
+  ==
+::
+```
+
+Here we have our `++on-scry` arm. The scry endpoints we've defined are divided
+into two parts: querying the update `$log` and retrieving entries from the
+`$journal`. Each end-point is as follows:
+
+- `/x/entries/all` - Retrieve all entries in the `$journal`. Our front-end will
+  use lazy-loading and only get a few at a time, so it won't use this. It's nice
+  to have it though, in case other agents want to get that data.
+
+- `/x/entries/before/[before]/[max]` - Retrieve at most `[max]` entries older
+  than the entry on `[before]` date. This is so our lazy-loading front-end can
+  progressively load more as the user scrolls down the page. The Javascript
+  front-end will format numbers without dot separators, so the path will look
+  like `/x/entries/before/1648051573109/10`. We therefore have to use the
+  [`++dem`](docs/hoon/reference/stdlib/4i#dem) parsing `rule` in a
+  [`++rash`](/docs/hoon/reference/stdlib/4g#rash) parser to convert it to an
+  ordinary atom. We then use the `++tap:log-orm` `mop` function to retrieve the
+  requested range as a list and return it as an `$update` with a
+  `%journal-update` mark.
+
+- `/x/entries/between/[start]/[end]` - Retrieve all journal entries between two
+  dates. This is so our front-end can have a search function, where the user can
+  enter a start and end date and get all the entries in between. The
+  `++lot:j-orm` `mop` function returns the subset of a `mop` between the two
+  given keys as a `mop`, and then we call `++tap:j-orm` to convert it to a list.
+  The `++lot:j-orm function` excludes the start and end values, so we subtract 1
+  from the start and add 1 to the end to make sure it includes the full range.
+
+- `/x/updates/all` - Retrieve the entire update `$log`. Our front-end won't use
+  this but it might be useful for other agents, so we've included it here.
+
+- `/x/updates/since/[since]` - Retrieve all `$update`s that have happened since
+  the specified timestamp, if any. This is so our front-end (or another agent)
+  can resynchronize its state in the event its subscription is interrupted,
+  without having to fetch everything from scratch again.
+
+We don't use any of the other agent arms, so the remainder have all been passed
+to `default-agent` for handling:
+
+```hoon
+++  on-leave  on-leave:def
+++  on-agent  on-agent:def
+++  on-arvo   on-arvo:def
+++  on-fail   on-fail:def
+--
+```
```

Comment:
Do you include the entire files anywhere?  Would be nice to have in case one misses copy-paste on one set of arms.